### PR TITLE
auhentication added

### PR DESCRIPTION
**Pull request by rahulyadav139**
Demo: https://statuesque-creponne-891db8.netlify.app/

### CODE REVIEW

File: src/components/authorization/LoginForm.js [hunk 1]
```diff
@@ -35,8 +39,38 @@ const LoginForm = props => {
...
+  const submitHandler = async e => {
+    e.preventDefault();
+    if (!emailIsValid || !passwordIsValid) {
+      emailIsTouched(true);
+      passwordIsTouched(true);
+      return;
+    }
+
+    const { data, status, error } = await sendData(
+      'http://localhost:8080/auth/login',
```
Comment (j24m) [inline src/components/authorization/LoginForm.js:51]: Suggestion: You can save backend URL in .env file.

File: src/components/authorization/LoginForm.js [hunk 2]
```diff
@@ -35,8 +39,38 @@ const LoginForm = props => {
...
+      'POST',
+      { email, password },
+      false
+    );
+
+    if (error) return;
+
+    if (status === 404) {
+      return console.log('user not found');
```
Comment (j24m) [inline src/components/authorization/LoginForm.js:60]: Suggestion: Instead of console.log you can use toast for showing user errors.

**Review by j24m — APPROVED** [review]

File: src/App.js [hunk 3]
```diff
@@ -11,23 +11,27 @@ import {
...
-        <Route path="/profile" element={<Profile />} />
-        <Route path="/playlists" element={<Playlists />} />
-        <Route path="/playlist/:playlistName" element={<SinglePlaylist />} />
-        <Route path="/history" element={<History />} />
-        <Route path="/watch-later" element={<WatchLater />} />
+        {isAuth && <Route path="/profile" element={<Profile />} />}
+        {isAuth && <Route path="/playlists" element={<Playlists />} />}
+        {isAuth && (
+          <Route path="/playlist/:playlistName" element={<SinglePlaylist />} />
+        )}
```
Comment (mehulsatardekar) [inline src/App.js:31]: Nice work on handling the routes

**Review by mehulsatardekar — APPROVED** [review]

File: src/components/authorization/LoginForm.js (hunk 1 above)
Comment (rahulyadav139) [reply src/components/authorization/LoginForm.js:51]: Thank you for the suggestion.

**Pull request merged**